pkg/controller/nodes/nodes_controller.go

```diff
@@ -87,17 +100,39 @@ func (r *ReconcileNodes) Start(stop <-chan struct{}) error {
 	}
 }
 
+func (r *ReconcileNodes) onUpdate(nodeName string) error {
+	logger := r.logger.WithValues("node", nodeName)
+	logger.Info("node update notification received")
+
+	c, err := r.getCache()
+	if err != nil {
+		return err
+	}
+
+	err = r.updateNode(c, nodeName)
+	if err != nil {
```

Comment:
When you aren't defining new variables, I suggest putting the operation inside the if.

It depends on the case, but if it's just for error validation, it saves vertical space.

```suggestion
	if err = r.updateNode(c, nodeName); err != nil {
```